subtests/docker_cli/systemd/systemd.py

```diff
@@ -0,0 +1,81 @@
+"""
+Summary
+-------
+
+Operational Summary
+---------------------
+    1. Copy Unit file
+    2. Build image using predefined Dockerfile & p4321-server.py
+    3. Run image
+    4. Confirm p4321-server script is running
+    5. Stop container
+    6. cleanup
+
+"""
+
+from dockertest import subtest
+from shutil import copy
+from os.path import join
+from autotest.client import utils
+import time
+from dockertest.dockercmd import DockerCmd
+
+
+class systemd(subtest.Subtest):
+
+    def initialize(self):
+        super(systemd, self).initialize()
+        sysd_unitf = join(self.bindir, self.config['sysd_unit_file'])
+        dockerfilesrc = join(self.bindir, 'Dockerfile')
+        script_src = join(self.bindir, 'p4321-server.py')
+        dockerbuild_dir = join(self.bindir, self.config['tmpblddir'])
+        self.config['dockerbuild_dir'] = dockerbuild_dir
+        self.config['unitf_dest'] = join(self.config['sysd_dest'],
+                                        self.config['sysd_unit_file'])
+        utils.run("mkdir %s" %dockerbuild_dir, ignore_status=True)
+        copy(sysd_unitf, self.config['sysd_dest'])
+        copy(dockerfilesrc, dockerbuild_dir)
+        copy(script_src, dockerbuild_dir)
+        self.build_image()
+
+    def build_image(self):
+        # FIXME
+        # a temporary ugly hack
+        # may able to use build_base class
+        # dkrcmd = DockerCmd(self, 'build', self.config['build_options_csv'])
+        # self.config['dkrcmd'] = dkrcmd
+        # dkrcmd.execute()
+        # 
+        build_args = self.config['docker_path'] + " build --force-rm -t " \
+                    + self.config['name'] + " -f " \
+                    + self.config['dockerbuild_dir'] +"/Dockerfile " \
+                    + self.config['dockerbuild_dir']
+        utils.run(build_args)
+
+    def run_once(self):
+        self.sysd_action(action = 'daemon-reload', sysd_unit = 'None')
+        self.sysd_action(action = 'start',
+                            sysd_unit = self.config['sysd_unit_file'])
+        # FIXME
+        # instead of sleep, use (not utils.is_port_free(4321, localhost))
+        time.sleep(5)
+        system_time = utils.run("cat < /dev/tcp/localhost/4321",
+                                ignore_status=True)
+        system_time_adigit = system_time.stdout.strip().isdigit() 
+        self.failif(not system_time_adigit,
+                    "Returned system time is not a digit")
+        # stop container
+        self.sysd_action(action = 'stop',
+                            sysd_unit = self.config['sysd_unit_file'])
+
+    def sysd_action(self, action = None, sysd_unit = None):
+        if sysd_unit is 'None':
```

Comment:
This does not do what you think it does! You are comparing to the string 'None', not to the actual python None. Huge difference. See below for a way to avoid this confusion.